projects/eslint-plugin/rules/strict-doc-example-extensions.js

```diff
@@ -0,0 +1,69 @@
+const getTypeName = require('../utils/get-type-name');
+
+const MESSAGE_ID = `strict-doc-example-extensions`;
+const ERROR_MESSAGE = `This mapping has wrong extension`;
+
+const DOC_EXAMPLE_INTERFACE_NAME = 'TuiDocExample';
+
+/**
+ * @type {Record<string, string>}
+ */
+const fileNameToExtension = {
+    TypeScript: '.ts',
+    HTML: '.html',
+    LESS: '.less',
+};
+
+/**
+ * @type {import('eslint').Rule.RuleModule}
+ */
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {description: ERROR_MESSAGE},
+        messages: {
+            [MESSAGE_ID]: ERROR_MESSAGE,
+        },
+        fixable: 'code',
+    },
+    create(context) {
+        return {
+            /**
+             * It would be better to support any object expression, not only within a class,
+             * but it's pretty complicated to extract the object type since all parents have different interfaces.
+             */
+            'PropertyDefinition[value.type="ObjectExpression"]'(node) {
```

Comment:
Let me know if you also want to support non class properties. I'll try to figure out how to handle

Reply:
`non class properties.` not now